src/coreclr/tests/testenvironment.proj

```diff
@@ -176,13 +176,31 @@
   <Target Name="CreateTestEnvFile" DependsOnTargets="OutputCOMPlusVariables">
     <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
       <_TestEnvFileLine Include="@(_COMPlusVariable->'set %(Identity)=%(Value)')"  />
+
+      <!-- Mono interpreter -->
       <_TestEnvFileLine Include="set MONO_ENV_OPTIONS=--interpreter" Condition="'$(Scenario)' == 'interpreter'" />
+
+      <!-- CLR interpreter -->
```

Comment:
This looks good to me, although `InterpreterJITThreshold` seems unnecessary here because we have enabled `ForceInterpreter`. BTW, if `ForceInterpreter` is not set, then `InterpreterJITThreshold` and `TC_CallCountThreshold` should also be set because the latter also controls the transition between tiers.

I am not sure whether `ZapDisable` should be set to `1`. It seems that `ZapDisable=0` with `ForceInterpreter=1` also triggers the interpreter. Do you have any thought for this?

In addition, should we setup a test environment so that we can also test the transition from the interpreter to Tier 1?

Thanks!

Reply:
It looks like `InterpreterJITThreshold` is used to trigger JIT compilation, but if you set `ForceInterpreter=1` the resulting JIT'ed functions are never used. So setting it to a high number just avoids the overhead of the JIT.

It also seems like there are two mechanisms for moving from interpreter to tier-1: both the normal tiering mechanism, as well as `Interpreter::JitMethodIfAppropriate()` (which still uses the tiering mechanism). I wonder if that's historical, since the interpreter "invented" the tiering long before the VM tiering system was written.

It seems like maybe we should _not_ set `ForceInterpreter` and thus let the interpreter participate in tiering. We can set the tiering threshold higher to get more interpreter coverage before tiering kicks in. However, we will miss some coverage, as there appear to be some functions which are never interpreted in normal `Interpreter=*` mode. E.g., it looks like PInvoke stubs force full optimization (probably via `MethodDesc::DetermineAndSetIsEligibleForTieredCompilation()`) -- I recall they don't participate in tiering. That's probably a fine trade-off.

Setting `ZapDisable=1` probably does nothing. But `COMPlus_ReadyToRun=0` would force interpret/JIT of all precompiled images. That might be interesting to test, but also presumably would make everything run very slowly.